src/prefect/cli/deployment.py

```diff
@@ -1309,3 +1316,6 @@ def cast_value(value: str) -> Any:
             )
 
     return parsed
+
+def timeout_handler(signum, frame):
+    raise Exception("Network not connected, please check connectivity on port 4200")
```

Comment:
I think this can be improved using a more specific `Exception` class. It's also not guaranteed that we're trying to connect on port 4200 -- we can include information on the address we connected to instead (using the `PREFECT_API_URL` variable)